db/error_handler.cc

```diff
@@ -381,4 +393,122 @@ Status ErrorHandler::RecoverFromBGError(bool is_manual) {
   return bg_error_;
 #endif
 }
+
+Status ErrorHandler::StartRecoverFromRetryableBGIOError(IOStatus io_error) {
+#ifndef ROCKSDB_LITE
+  db_mutex_->AssertHeld();
+  if (bg_error_.ok() || io_error.ok()) {
+    return Status::OK();
+  }
+  if (db_options_.max_bgerror_resume_count <= 0 || recovery_in_prog_ ||
+      recovery_thread_) {
+    // Auto resume BG error is not enabled, directly return bg_error_.
+    return bg_error_;
+  }
+
+  recovery_in_prog_ = true;
+  recovery_thread_.reset(
+      new port::Thread(&ErrorHandler::RecoverFromRetryableBGIOError, this));
```

Comment:
Imagine a process with many RocksDB instances. Is it possible that the underlying system starts to return retryable errors to each of them from a certain time? If so, the process will spawn a large number of such unmanaged threads, which can be a concern.

Reply:
@riversand963 Yeah. Good point. We have the discussion before about this concern. The decision is, in the current stage, it is ok to have one thread for one DB (for some cases, might be hundred DB in a host), since all DB is in read mode. No flush or compaction will be scheduled. But in the next step, we will put them in a shared thread pool or other way to solve this potential isssue.